### ISSUE TITLE
triple-slash-reference gives false positives for d.ts files

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_lint#461]
```
// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.

/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any, no-var */

/// <reference no-default-lib="true" />
/// <reference lib="esnext" />

...
```

```
error[triple-slash-reference]: `triple slash reference` is not allowed
 --> /Users/biwanczuk/dev/deno/op_crates/fetch/lib.deno_fetch.d.ts:7:1
  |
7 | /// <reference lib="esnext" />
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
```

We must ensure that declaration files are properly linted. Ref #305

**Comment by bartlomieju** [comment]
I'm a bit puzzled at this rule now... @kitsonk does it even make sense in context of Deno to prohibit tripe slash references?

**Comment by kitsonk** [comment]
There are a lot of cases where they shouldn't be used but several cases where they should be used. A linter isn't going to be able to tell that. Best to be left to the documentation.
